File: node/create_domain.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  try {
    let identity = await dnsimple.identity.whoami();
    const accountID = identity.data.account.id;
    const domainName = process.env.DOMAIN;

    if (!domainName) {
      console.error('Please specify a domain name to create');
      return;
    }

    const domain = await dnsimple.domains.createDomain(accountID, { name: domainName });

    console.log(domain);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();

File: elixir/create_domain.exs
# Usage: mix run create_domain.exs example.com

HTTPoison.start

client = %Dnsimple.Client{
  access_token: Application.get_env(:dnsimple, :access_token),
  base_url: "https://api.sandbox.dnsimple.com/"
}

{:ok, response} = Dnsimple.Identity.whoami(client)
account_id = response.data.account["id"]

name = List.last(System.argv)
IO.puts "Create domain #{name}"

IO.inspect Dnsimple.Domains.create_domain(client, account_id, %{name: name})
